tools/README.md

```diff
@@ -112,115 +77,199 @@ export repo_owner=<your github account>
+
+| Name | Description |
+| --- | --- |
+| [Jenkinsfile](https://github.com/bcgov/nr-showcase-devops-tools/blob/master/Jenkinsfile) | this is the master branch build pipeline |
+| [Jenkinsfile.cicd](https://github.com/bcgov/nr-showcase-devops-tools/blob/master/Jenkinsfile.cicd) | this is the Pull Request build pipeline |
+| [sonar-project.properties](https://github.com/bcgov/nr-showcase-devops-tools/blob/master/sonar-project.properties) | this is will configure your SonarQube scans |
+
+This repository contains skeleton examples, see projects [NR Get Token](https://github.com/bcgov/nr-get-token.git) and [NR Messaging Service Showcase](https://github.com/bcgov/nr-messaging-service-showcase.git) for full examples.
+
+A very simple example to demonstrate a build and SonarQube analysis (no additional setup, no servers, no deployments), can be found at [nr-showcase-devops-tools-demo-sq](https://github.com/bcgov/nr-showcase-devops-tools-demo-sq.git).  This can be forked and used to test the Jenkins and SonarQube installation, and to demonstrate the master and pull request jobs.  For instance, the master branch will show Code Smells and duplicates in SonarQube.  In you fork, create a pull request that corrects these and see how the PR Jenkins job kicks off.
+
+### Jenkins Server Setup
+
+The following commands setup up Jenkins and uses this repository and specific OpenShift project namespaces.
+
+#### Create Secrets
+
+The BCDevOps CICD Jenkins Basic install requires a template github secret and a template for the slave.  This will create the secrets named as it requires.
 
 ```sh
-curl https://raw.githubusercontent.com/BCDevOps/sonarqube/bbb9f62e29706b61382cf24d7ad7e08f2476a01f/provisioning/updatesqadminpw.sh > updatesqadminpw.sh
+oc -n $tools process -f "$templates_url/secrets.yaml" -p GH_USERNAME=$gh_username -p GH_PASSWORD=$gh_password | oc  -n $tools create -f -
 ```
 
-Then simply run the following script and follow its instructions. Make sure you save the new password in an OpenShift secret or equivalent!  Ensure that the SonarQube app is fully deployed and operational before running this script.
+#### Create ConfigMap for related namespaces
+
+For our custom jobs scripts and Jenkinsfiles.
 
 ```sh
-chmod +x updatesqadminpw.sh && ./updatesqadminpw.sh
+oc -n $tools process -f "$templates_url/ns-config.yaml" -p DEV=$dev -p TEST=$test -p PROD=$prod -p TOOLS=$tools | oc  -n $tools create -f -
 ```
 
-Go to https://sonarqube-<$tools>.pathfinder.gov.bc.ca and log in as `admin` with the new password (it is stored in sonarqube-admin-password secret).
+#### Create ConfigMap for the application
 
-## Jenkins
+For our custom jobs scripts and Jenkinsfiles.
+
+```sh
+oc -n $tools process -f "$templates_url/jobs-config.yaml" -p REPO_OWNER=$repo_owner -p REPO_NAME=$repo_name -p APP_NAME=$app_name -p APP_DOMAIN=$app_domain | oc -n $tools create -f -
```

Comment:
-l app=jenkins-prod

Reply:
The app label is embedded into the template now